File: mlir/examples/dsp/SimpleBlocks/Output/TryDSPApps/BenchmarkTest/DSP-DSL/radarSignalProcessing.py
def main() {

        var antennas = 4;
        var input_fc = 5;
        var N = 101;
	var input = getRangeOfVector(0, 100000000, 0.000125);
        var weights = getRangeOfVector(-90, 180, 1);

        var signal = beam_form(antennas, input_fc, input, weights);
        var b1 = abs(signal);
        var power_profile = b1 * b1;
        var power_angle_max_idx = argmax(power_profile, 0);
        var power_angle_max_ele = argmax(power_profile,0);

        var pi = 3.1415926;
        var fc1 = 1000;
        var fc2 = 7500;
        var Fs = 8000;

        var wc1 = 2*pi*fc1 / Fs;
        var filter1 = lowPassFIRFilter(wc1, N);
        var filter_hamming_1 = filter1 * hamming(N);
        var wc2 = 2*pi*fc2 / Fs;
        var filter2 = highPassFIRFilter(wc2, N);
        var filter_hamming_2 = filter2 * hamming(N);

        var bpf = sub(filter_hamming_2, filter_hamming_1);
        var firFilterResponse = FIRFilterResponse(power_profile, bpf);
        var final = getElemAtIndx(firFilterResponse , 2);
        print(final);
}

File: mlir/examples/dsp/SimpleBlocks/Output/TryDSPApps/BenchmarkTest/Matlab/radarSignalProcessing.m
% Constants
PI = pi;
INPUT_LENGTH = 10000;

% Function prototypes
input = linspace(0, (INPUT_LENGTH - 1) * 0.000125, INPUT_LENGTH); % Using linspace instead of getrangeofvector
weights = linspace(-90, 180, 4); % Example antenna weights
antennas = 4;
input_fc = 5;
N = 101;
fc1 = 1000;
fc2 = 7500;
Fs = 8000;

% Generate beamformed signal
signal = beamForm(antennas, input_fc, input, weights);

% Compute absolute values and power profile
b1 = abs(signal);
power = b1 .^ 2; % element-wise square instead of power_profile

% Low-pass and high-pass FIR filters with Hamming window
wc1 = 2 * PI * fc1 / Fs;
filter1 = lowPassFIRFilter(wc1, N);
filter_hamming_1 = filter1 .* hamming(N, 'symmetric')'; % Using 'symmetric' Hamming window

wc2 = 2 * PI * fc2 / Fs;
filter2 = highPassFIRFilter(wc2, N);
filter_hamming_2 = filter2 .* hamming(N, 'symmetric')'; % Using 'symmetric' Hamming window

% Band-pass filter by subtracting the filters
bpf = filter_hamming_2 - filter_hamming_1;

% Apply FIR filter to the power profile (use full convolution)
firFilterResponse = conv(power, bpf, 'full'); % Use 'full' to match C code

% Output final value at the 10000th index (adjust if necessary)
final = firFilterResponse(2); % Adjust to match desired index in C code
fprintf('final: %f\n', final);

% Functions

function output = beamForm(antennas, frequency, time, weights)
    phase_var = 2 * pi * frequency;
    signal = zeros(antennas, length(time));

    for i = 1:antennas
        iter_args = (i - 1) * pi / 4.0;
        signal(i, :) = sin(time * phase_var + iter_args);
    end

    output = sum(signal .* weights', 1); % Beamforming by weighted summation
end

function output = lowPassFIRFilter(wc, N)
    midIndex = (N - 1) / 2;
    output = zeros(1, N);

    for i = 1:N
        if i == midIndex + 1
            output(i) = wc / pi;
        else
            output(i) = sin(wc * (i - midIndex - 1)) / (pi * (i - midIndex - 1));
        end
    end
end

function output = highPassFIRFilter(wc, N)
    midIndex = (N - 1) / 2;
    output = zeros(1, N);

    for i = 1:N
        if i == midIndex + 1
            output(i) = 1 - wc / pi;
        else
            output(i) = -sin(wc * (i - midIndex - 1)) / (pi * (i - midIndex - 1));
        end
    end
end
